src/Modules/OrchardCore.Commerce/Controllers/PaymentController.cs

```diff
@@ -133,6 +142,27 @@ public async Task<IActionResult> SuccessPost(string orderId)
         if (await _contentManager.GetAsync(orderId) is not { } order) return NotFound();
 
         await _contentItemDisplayManager.UpdateEditorAsync(order, _updateModelAccessor.ModelUpdater, isNew: false);
+
+        // Saving addresses.
+        var userService = _userServiceLazy.Value;
+        var orderPart = order.As<OrderPart>();
+
+        if (await userService.GetCurrentFullUserAsync(_hca) is { } user)
```

Comment:
You don't need to inject HCA in a controller, use `userService.GetFullUserAsync(User)`.